addons/website_event_booth_sale/controllers/event_booth.py

```diff
@@ -20,20 +21,29 @@ def event_booth_main(self, event):
     @http.route()
     def event_booth_registration_confirm(self, event, booth_category_id, event_booth_ids, **kwargs):
         booths = self._get_requested_booths(event, event_booth_ids)
-        booth_category = request.env['event.booth.category'].sudo().browse(int(booth_category_id))
-        order = request.website.sale_get_order(force_create=1)
-        order._cart_update(
-            product_id=booth_category.product_id.id,
-            set_qty=1,
-            event_booth_pending_ids=booths.ids,
-            registration_values=self._prepare_booth_registration_values(event, kwargs),
-        )
-        if order.amount_total:
-            return request.redirect('/shop/checkout')
-        elif order:
-            order.action_confirm()
-            request.website.sale_reset()
-
-        return request.redirect(('/event/%s/booth/success?' % event.id) + werkzeug.urls.url_encode({
-            'booths': ','.join([str(id) for id in booths.ids]),
-        }))
+
+        if booths:
+            booth_category = request.env['event.booth.category'].sudo().browse(int(booth_category_id))
```

Comment:
Could put this before the "if booth" check and also return an error `if not booth_category.exists()`.